utils/generate_vim_syntax.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+# Copyright (c) 2016 Google Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates Vim syntax rules for SPIR-V assembly (.spvasm) files"""
+
+from __future__ import print_function
+
+import json
+
+PREAMBLE="""" Vim syntax file
+" Language:   spvasm
+" Generated by SPIRV-Tools
+
+if version < 600
+  syntax clear
+elseif exists("b:current_syntax")
+  finish
+endif
+
+syn case match
+"""
+
+POSTAMBLE="""
+
+syntax keyword spvasmTodo TODO FIXME contained
+
+syn match   spvasmIdNumber /%\d\+\>/
+
+" The assembler treats the leading minus sign as part of the number token.
+syn match   spvasmNumber /-\?\<\d\+\>/
+syn match   spvasmFloat  /-\?\<\d\+\.\d*\(e[+-]\d\+\)\?\>/
+
+" Hex floats.
+" In general, C++ requires at least one hex digit, and the decimal
+" point is optional.
+" First case: hex digits before the optional decimal, no trailing hex digits.
+syn match   spvasmFloat  /-\?0[xX]\\x\+.*p[-+]\d\+\>/
+" Second case: optional hex digits before decimal, trailing hex digits
+syn match   spvasmFloat  /-\?0[xX]\\x*.\\x\+p[-+]\d\+\>/
+
+syn match   spvasmComment /;.*$/ contains=spvasmTodo
+syn region  spvasmString start=/"/ skip=/\\\\"/ end=/"/
+syn match   spvasmId /%[a-zA-Z_][a-zA-Z_0-9]*/
+
+" Highlight unknown constants and statements as errors
+syn match   spvasmError /[a-zA-Z][a-zA-Z_0-9]*/
+
+
+if version >= 508 || !exists("did_c_syn_inits")
+  if version < 508
+    let did_c_syn_inits = 1
+    command -nargs=+ HiLink hi link <args>
+  else
+    command -nargs=+ HiLink hi def link <args>
+  endif
+
+  HiLink spvasmStatement Statement
+  HiLink spvasmNumber Number
+  HiLink spvasmComment Comment
+  HiLink spvasmString String
+  HiLink spvasmFloat Float
+  HiLink spvasmConstant Constant
+  HiLink spvasmIdNumber Identifier
+  HiLink spvasmId Identifier
+  HiLink spvasmTodo Todo
+
+  delcommand HiLink
+endif
+
+let b:current_syntax = "spvasm"
+"""
+
+# This list is taken from the description of OpSpecConstantOp in SPIR-V 1.1.
+SPEC_CONSTANT_OP_OPCODES = """
```

Comment:
Maybe we should extend the grammar file with a field for each opcode to indicate whether it's possible to be used for specConstantOp.

I generally think the more information we put into the grammar file, the more useful it will be.